openwisp_radius/api/views.py

```diff
@@ -294,16 +295,23 @@ def get_user(self, serializer, *args, **kwargs):
 
     def validate_membership(self, user):
         if not (user.is_superuser or user.is_member(self.organization)):
-            if OrganizationUser.objects.filter(user=user).exists():
+            try:
+                org_registration_enabled = (
+                    self.organization.radius_settings.get_registration_enabled()
+                )
+            except OrganizationRadiusSettings.DoesNotExist:
+                org_registration_enabled = app_settings.REGISTRATION_API_ENABLED
```

Comment:
let's avoid this duplication between here and the changes in openwisp_radius/api/permissions.py, see how something similar is done with a function: https://github.com/openwisp/openwisp-radius/blob/master/openwisp_radius/api/utils.py#L43-L52.